src/server.c

```diff
@@ -5283,6 +5283,17 @@ int processCommand(client *c) {
         return C_OK;
     }
 
+    /* check if DEBUG/MODULE command not allowed */
+    if ((c->cmd->proc == debugCommand && !allowProtectedAction(server.enable_debug_cmd, c)) ||
+        (c->cmd->proc == moduleCommand && !allowProtectedAction(server.enable_module_cmd, c))) {
```

Comment:
Why are we blocking "module list". It's possible a module could be loaded at startup, which isn't a changed code path here, but clients won't be able to list them.

Reply:
you're right,
let's add a command flag, and avoid flagging the "read-only" MODULE sub-commands.